src/script/microbit-interfacing/MicrobitSerial.ts

```diff
@@ -12,17 +12,27 @@ import * as protocol from './serialProtocol';
 
 const enum SerialProtocolState {
   AwaitingHandshakeResponse,
+  AwaitingConfiguration,
+  Ready,
   Running,
 }
 
 class MicrobitSerial implements MicrobitConnection {
   private serialProtocolState = SerialProtocolState.AwaitingHandshakeResponse;
-  private unprocessedInput = '';
+  // TODO: The radio frequency should be randomly generated once per session.
+  //       If we want a session to be restored (e.g. from local storage) and
+  //       the previously flashed micro:bits to continue working without
+  //       reflashing we need to store and retrieve this value somehow.
+  private static sessionRadioFrequency = -1;
```

Comment:
We need to clarify if this is a requirement, if it is we can always implement that in a future PR, but happy to do that here if I could get some guidance on how to hook to the app session management.